### PR TITLE
[MM-58284] Confirmation when leaving a call

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Added confirmations on the widget, global widget, expanded view, and post component's leave call buttons
- The global widget required an added shim to give room for the confirmation, please see if it works well on linux desktop :)
- The popup system needed a bit of reworking to add the tooltips, kind of a mess, but I think I figured out how to do it in an elegant way that doesn't affect the existing uses (host controls), but would appreciate a second pair of eyes on that just in case (that the tooltips look correct in all places on the leave button, and that the host controls still work/look as expected).

First row is from desktop and host, second row is from participant and webapp:

| | | |
|--|--|--|
| image | image | image |
| image | image | image | 

- Designs: https://www.figma.com/design/sECejPEfl6OjuJxj0WU9sb/MM-58284-Leaving-and-Ending-a-Call?node-id=1-7&m=dev

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-58284

### CODE REVIEW

**Review by streamer45 — COMMENTED** [review]
Looks great, even on :penguin: :)

Two things I noticed:

- The keyboard shortcut causes the user to leave immediately. This is fine but inconsistent with how we are handling a similar case in https://github.com/mattermost/mattermost-plugin-calls/pull/760.
- Ending the call is not working as I'd expect, although this is unrelated to this PR. The call post should update immediately but for some reason, that's not happening until the server timeout triggers. We should look into this.

File: webapp/src/components/expanded_view/component.tsx [hunk 1]
```diff
@@ -1415,7 +1417,6 @@ const MuteOthersButton = styled.button`
     i {
         font-size: 14px;
     }
->>>>>>> origin/main
```
[the remark this answers is not in the document]
Comment (cpoile) [reply webapp/src/components/expanded_view/component.tsx:1418]: I don't know how that got there and how it survived.

**Comment by cpoile** [comment]
@streamer45 for 1: Earlier I had messaged @abhijit-singh about it too. I suggested we keep the shortcut as it has been, under the assumption that if you hit it that means you want it to happen right then, not that you then want to use the mouse and click again. Your right that it doesn't match the end recording, but in that case the action is more destructive, whereas you can always rejoin if you left by accident. Imo, it just seems annoying that the shortcut just opens a little popup. If we made it do that, I would vote we just remove the shortcut altogether for uselessness. :)

For 2, I can look into that. I think it's been happening like that for a long time.

**Comment by streamer45** [comment]
> @streamer45 for 1: Earlier I had messaged @abhijit-singh about it too. I suggested we keep the shortcut as it has been, under the assumption that if you hit it that means you want it to happen right then, not that you then want to use the mouse and click again. Your right that it doesn't match the end recording, but in that case the action is more destructive, whereas you can always rejoin if you left by accident. Imo, it just seems annoying that the shortcut just opens a little popup. If we made it do that, I would vote we just remove the shortcut altogether for uselessness. :)

Yeah, I agree. Shortcut to show a popup is weird.

> For 2, I can look into that. I think it's been happening like that for a long time.

Thanks.

**Review by streamer45 — APPROVED** [review]
Great! And thanks for the brave effort on e2e and related lint checks.

**Comment by cpoile** [comment]
@streamer45 Fixed how we were doing call end, and exposing the `end call for everyone` button to admins.

**Review by streamer45 — APPROVED** [review]
Working great now!

**Comment by cpoile** [comment]
fyi: added a rule to only show `End call for everyone` when there are other people in the call.

**Review by abhijit-singh — CHANGES_REQUESTED** [review]
Thanks @cpoile! Looks good! I have a few requests - 
1. In the expanded window, can we align the right edge of the confirmation pop-out with the right edge of the button?
2. Also in the expanded window, can we make the border radius for the pop-out 8px to match that of the button?

Current: 
image

Ideal:
image

Note that the above two requests apply to the popout window only. 
3. Can we increase the width of the menu items to 36px to be consistent with the rest of the app? Right now it seems to be 32px. 

Current:
image

4. On the Desktop App, in the floating widget, clicking outside the popout menu (within the widget boundries, does not close the popout menu. Ideally it should close when clicked outside.

**Comment by cpoile** [comment]
@abhijit-singh 1-3 done, thanks! 4 took me awhile, but I don't think we can solve it. The problem is that when you click on another part of the widget that is meant to be draggable, the click doesn't register (but you can drag). If you click on any place you can click, the menu will close (e.g. on a participant in the participant list, or any of the widget buttons). So unfortunately I think we're stuck with that -- the user will have to click on the `Cancel` button if they change their mind (or click on a participant or settings menu item).

**Review by abhijit-singh — APPROVED** [review]
Looks good to me, thanks @cpoile!

**Pull request merged**